keras/src/export/litert.py

```diff
@@ -0,0 +1,255 @@
+
+
+def export_litert(
+    model,
+    filepath,
+    input_signature=None,
+    **kwargs,
+):
+    """Export the model as a LiteRT artifact for inference.
+
+    Args:
+        model: The Keras model to export.
+        filepath: The path to save the exported artifact.
+        input_signature: Optional input signature specification. If
+            `None`, it will be inferred.
+        **kwargs: Additional keyword arguments passed to the exporter.
+    """
+
+    exporter = LiteRTExporter(
+        model=model,
+        input_signature=input_signature,
+        **kwargs,
+    )
+    exporter.export(filepath)
+    io_utils.print_msg(f"Saved artifact at '{filepath}'.")
+
+
+class LiteRTExporter:
+    """Exporter for the LiteRT (TFLite) format.
+
+    This class handles the conversion of Keras models for LiteRT runtime and
+    generates a `.tflite` model file. For efficient inference on mobile and
+    embedded devices, it creates a single callable signature based on the
+    model's `call()` method.
+    """
+
+    def __init__(
+        self,
+        model,
+        input_signature=None,
+        **kwargs,
+    ):
+        """Initialize the LiteRT exporter.
+
+        Args:
+            model: The Keras model to export
+            input_signature: Input signature specification (e.g., TensorFlow
+                TensorSpec or list of TensorSpec)
+            **kwargs: Additional export parameters
+        """
+        self.model = model
+        self.input_signature = input_signature
+        self.kwargs = kwargs
+
+    def _infer_dict_input_signature(self):
+        """Infer input signature from a model with dict inputs.
+
+        This reads the actual shapes and dtypes from model._inputs_struct.
+
+        Returns:
+            dict or None: Dictionary mapping input names to InputSpec, or None
+        """
+        # Check _inputs_struct first (preserves dict structure)
+        if hasattr(self.model, "_inputs_struct") and isinstance(
+            self.model._inputs_struct, dict
+        ):
+            return {
+                name: make_input_spec(inp)
+                for name, inp in self.model._inputs_struct.items()
+            }
+
+        # Fall back to model.inputs if it's a dict
+        if hasattr(self.model, "inputs") and isinstance(
+            self.model.inputs, dict
+        ):
+            return {
+                name: make_input_spec(inp)
+                for name, inp in self.model.inputs.items()
+            }
+
+        return None
+
+    def export(self, filepath):
+        """Exports the Keras model to a TFLite file.
+
+        Args:
+            filepath: Output path for the exported model
+
+        Returns:
+            Path to exported model
+        """
+        # 1. Resolve / infer input signature
+        if self.input_signature is None:
+            # Try dict-specific inference first (for models with dict inputs)
+            dict_signature = self._infer_dict_input_signature()
```

Comment:
You are checking dict vs. not, but it's a bit more complicated than that. Inputs can be nested to any level, it can be a list of dicts of inputs, or a dict of lists, etc. But the good news is that it's actually easy to generalize with `keras.tree`, in fact, it will be less code once you handle any arbitrary structure.

There are 3 case:
1. single input
2. flat list of inputs
3. everything else (nested lists, dict...)

The way you check is:
```python
if not keras.tree.is_nested(inputs):
    # case 1
elif isintance(inputs, list) and len(inputs) == len(keras.tree.flatten(inputs)):
    # case 2
else:
    # case 3
``` 

You'll do this instead of the current `has_dict_inputs`.